private/pkg/protosource/protosource.go

```diff
@@ -841,6 +841,44 @@ func PackageToNameToService(files ...File) (map[string]map[string]Service, error
 	return packageToNameToService, nil
 }
 
+// PackageToDirectlyImportedPackageToFileImports maps packages to directly imported packages
+// to the FileImports that import this package.
+//
+// For example, if package a imports package b via c/d.proto and c/e.proto, this will have
+// a -> b -> [c/d.proto, c/e.proto].
+//
+// A directly imported package will not be equal to the package, i.e. there will be no a -> a.
+//
+// Files with no packages are included with key "" to be consistent with other functions.
+func PackageToDirectlyImportedPackageToFileImports(files ...File) (map[string]map[string][]FileImport, error) {
+	filePathToFile, err := FilePathToFile(files...)
+	if err != nil {
+		return nil, err
+	}
+	packageToDirectlyImportedPackageToFileImports := make(map[string]map[string][]FileImport)
+	for _, file := range files {
+		pkg := file.Package()
+		directlyImportedPackageToFileImports, ok := packageToDirectlyImportedPackageToFileImports[pkg]
+		if !ok {
+			directlyImportedPackageToFileImports = make(map[string][]FileImport)
+			packageToDirectlyImportedPackageToFileImports[pkg] = directlyImportedPackageToFileImports
+		}
+		for _, fileImport := range file.FileImports() {
+			importedFile, ok := filePathToFile[fileImport.Import()]
+			if ok {
```

Comment:
nit: in-line this one.
```suggestion
			if importedFile, ok := filePathToFile[fileImport.Import()]; ok {
```

Reply:
Ah I'm dumb